include/sampling/random_point_generators_multithread.hpp

```diff
@@ -0,0 +1,134 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+// Contributed and/or modified by Konstantinos Pallikaris, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef SAMPLERS_RANDOM_POINT_GENERATORS_MULTITHREAD_HPP
+#define SAMPLERS_RANDOM_POINT_GENERATORS_MULTITHREAD_HPP
+
+#include <iostream>
+#include <omp.h>
+#include <unistd.h>
+
+template
+<
+    typename Walk
+>
+struct RandomPointGeneratorMultiThread
+{
+    template
+    <
+        typename Polytope,
+        typename Point,
+        typename PointList,
+        typename WalkPolicy,
+        typename RandomNumberGenerator,
+        typename Parameters
+    >
+    static void apply(Polytope& P,
+                      Point &p,   // a point to start
+                      unsigned int const& rnum,
+                      unsigned int const& walk_length,
+                      unsigned int const& num_threads,
+                      PointList &randPoints,
+                      WalkPolicy &policy,
+                      RandomNumberGenerator &rng,
+                      Parameters const& parameters)
+    {
+        typedef typename Walk::template thread_parameters
+        <
+            NT,
+            Point
+        > _thread_parameters;
+
+
+        omp_set_num_threads(num_threads);
+        unsigned int jj = 0, d= P.dimension(), m = P.num_of_hyperplanes;
+        std::vector<int> num_points_per_thread(num_threads, 0);
```

Comment:
What about this
```
std::vector<int> num_points_per_thread(rnum%num_threads, rnum/num_threads+1);
std::vector<int> a(num_threads - rnum%num_threads, rnum/num_threads);
num_points_per_thread.insert(num_points_per_thread.end(), a.begin(), a.end());
```
instead of the double loop? Seems simpler and more efficient.

Reply:
Thank you @vissarion. I've implemented this.